pkg/resource/api.go

```diff
@@ -174,8 +174,7 @@ func NewAPIUpdatingApplicator(c client.Client) *APIUpdatingApplicator {
 }
 
 // Apply changes to the supplied object. The object will be created if it does
-// not exist, or updated if it does. If the object does exist and no
-// ApplyOptions are passed, the update will be a no-op.
+// not exist, or updated if it does.
 func (a *APIUpdatingApplicator) Apply(ctx context.Context, o runtime.Object, ao ...ApplyOption) error {
 	m, ok := o.(metav1.Object)
```

Comment:
```suggestion
	m, ok := o.(Object)
```

`resource.Object` captures both `metav1.Object` and `runtime.Object` and you can use `m` instead of `o` in the rest of the function to not have to do the cast later when you do `SetResourceVersion`

Reply:
True. I was trying to keep it matching more closely to the patching applicator, but I really don't think that is too much of a concern at this point 👍 Will update

Reply:
Actually since we are only setting resource version we can use `m` as `metav1.Object`

Reply:
@muvaf pushed an update, could you indicate your approval still applies?

Reply:
> Actually since we are only setting resource version we can use m as metav1.Object

The underlying object of both `o` and `m` is same but I think it'd be more straight-forward if we used `m` instead of `o` right after the type cast. Setting the resource version on `m` but actually updating `o` could be confusing for example.

Reply:
Okay, I updated to match.